plugins/web/opentelemetry-plugin-react-load/src/BaseOpenTelemetryComponent.ts

```diff
@@ -0,0 +1,451 @@
+        return apply;
+      };
+    };
+  }
+
+  /**
+   * Patches the componentWillUnmount lifecycle method
+   */
+  private _patchComponentWillUnmount(){
+    return (original: ComponentWillUnmountFunction): ComponentWillUnmountFunction => {
+      const plugin = this;
+     
+      return function patchComponentWillUnmount(
+        this: React.Component,
+        ...args
+      ): void {
+        const parentData = plugin._getCreateParentSpan(this);
+        if(parentData){
+          parentData.name = AttributeNames.UNMOUNTING_SPAN;
+        }
+        const apply = plugin._instrumentFunction(this, 'componentWillUnmount', () => { return original!.apply(this, args)});
+        plugin._endParentSpan(this);
+        return apply;
+      };
+    };
+  }
+
+  /**
+   * patch function which wraps all the lifecycle methods
+   */
+  public patch() {
+    BaseOpenTelemetryComponent._logger.debug('applying patch to', this.moduleName, this.version);
+    
+    if (isWrapped(this.render)) {
+      shimmer.unwrap(this, 'render');
+      BaseOpenTelemetryComponent._logger.debug('removing previous patch from method render');
+    }
+    if (isWrapped(this.componentDidMount)) {
+      shimmer.unwrap(this, 'componentDidMount');
+      BaseOpenTelemetryComponent._logger.debug(
+        'removing previous patch from method componentDidMount'
+      );
+    }
+    if (isWrapped(this.shouldComponentUpdate)) {
+      shimmer.unwrap(this, 'shouldComponentUpdate');
+      BaseOpenTelemetryComponent._logger.debug('removing previous patch from method shouldComponentUpdate');
+    }
+    if (isWrapped(this.getSnapshotBeforeUpdate)) {
+      shimmer.unwrap(this, 'getSnapshotBeforeUpdate');
+      BaseOpenTelemetryComponent._logger.debug('removing previous patch from method getSnapshotBeforeUpdate');
+    }
+    if(isWrapped(this.setState)){
+      shimmer.unwrap(this, 'setState');
+      BaseOpenTelemetryComponent._logger.debug('removing previous patch from method setState');
+    }
+    if(isWrapped(this.forceUpdate)){
+      shimmer.unwrap(this, 'forceUpdate');
+      BaseOpenTelemetryComponent._logger.debug('removing previous patch from method forceUpdate');
+    }
+    if (isWrapped(this.componentDidUpdate)) {
+      shimmer.unwrap(this, 'componentDidUpdate');
+      BaseOpenTelemetryComponent._logger.debug('removing previous patch from method componentDidUpdate');
+    }
+    if (isWrapped(this.componentWillUnmount)) {
+      shimmer.unwrap(this, 'componentWillUnmount');
+      BaseOpenTelemetryComponent._logger.debug('removing previous patch from method componentWillUnmount');
+    }
+
+    // Lifecycle methods must exist when patching, even if not defined in component
+    if (!this.render) {
+      this.render = () => {
+        return null;
+      };
+    }
+    if (!this.componentDidMount) {
+      this.componentDidMount = () => {};
+    }
+    if (!this.shouldComponentUpdate) {
+      this.shouldComponentUpdate = () => { return true; };
```

Comment:
Should the default value for `shouldComponentUpdate` be true if it doesn't exist?

Reply:
When the function is not defined, it's actually never called. The default behaviour is equivalent to always updating when an update flow is started (as opposed to short circuiting and cancelling the update) which is equivalent to always returning true!